src/Console/Command/CompileCommand.php

```diff
@@ -0,0 +1,205 @@
+                $phpExtensions,
+                $input->getOption('php-version'),
+                $spcBinaryDir,
+                $io
+            );
+
+            $this->buildPHP(
+                $spcBinaryPath,
+                $phpExtensions,
+                $arch,
+                $spcBinaryDir,
+                $io
+            );
+        }
+
+        $this->mergePHPandPHARIntoSingleExecutable(
+            $spcBinaryPath,
+            $input->getArgument('phar-path'),
+            $input->getOption('output'),
+            $spcBinaryDir,
+            $io
+        );
+
+        return Command::SUCCESS;
+    }
+
+    private function downloadSPC(string $spcSourceUrl, string $spcBinaryDestination, OutputInterface $output): void
+    {
+        $response = $this->httpClient->request('GET', $spcSourceUrl);
+        $contentLength = $response->getHeaders()['content-length'][0] ?? 0;
+
+        $outputStream = fopen($spcBinaryDestination, 'w');
+        $progressBar = new ProgressBar($output, (int) $contentLength);
+
+        if (false === $outputStream) {
+            throw new \RuntimeException(sprintf('Failed to open file "%s" for writing', $spcBinaryDestination));
+        }
+
+        foreach ($this->httpClient->stream($response) as $chunk) {
+            fwrite($outputStream, $chunk->getContent());
+            $progressBar->advance(\strlen($chunk->getContent()));
+        }
+
+        fclose($outputStream);
+        chmod($spcBinaryDestination, 0o755);
+
+        $progressBar->finish();
+    }
+
+    private function installPHPBuildTools(string $spcBinaryPath, string $spcBinaryDir, SymfonyStyle $io): void
+    {
+        $installSPCDepsProcess = new Process(
+            command: [
+                $spcBinaryPath, 'doctor',
+                '--auto-fix',
+            ],
+            cwd: $spcBinaryDir,
+            timeout: 5 * 60
+        );
+        $io->text('Running command: ' . $installSPCDepsProcess->getCommandLine());
+        $installSPCDepsProcess->mustRun(fn ($type, $buffer) => print ($buffer));
+    }
+
+    private function downloadPHPSourceDeps(string $spcBinaryPath, mixed $phpExtensions, mixed $phpVersion, string $spcBinaryDir, SymfonyStyle $io): void
+    {
+        $downloadProcess = new Process(
+            command: [
+                $spcBinaryPath, 'download',
+                '--for-extensions=' . $phpExtensions,
+                '--with-php=' . $phpVersion,
+            ],
+            cwd: $spcBinaryDir,
+            timeout: 5 * 60
+        );
+        $io->text('Running command: ' . $downloadProcess->getCommandLine());
+        $downloadProcess->mustRun(fn ($type, $buffer) => print ($buffer));
+    }
+
+    private function buildPHP(string $spcBinaryPath, mixed $phpExtensions, mixed $arch, string $spcBinaryDir, SymfonyStyle $io): void
+    {
+        $buildProcess = new Process(
+            command: [
+                $spcBinaryPath, 'build', $phpExtensions,
+                '--build-micro',
```

Comment:
is --build-cli better?

Reply:
AFAIU, `--build-cli`  is meant to build a php binary exclusively, only the `micro` option allows to glue php binary and php source code into one file